File: meetings/2019-10-13/yunsoo.cpp
class Solution {
public:
    int magicalString(int n) {
        vector<int> s = {1, 2, 2};
        int toggle = 2;
        int s_idx = 2;
        int s_size = s.size();
        int k = 0;
        
        while(s_size < n){
            toggle ^= 3;
            k = s[s_idx];
            for(int i = 0; i < k; i++){
                s.push_back(toggle);
            }
            s_idx++;
            s_size += k;
        }
        
        int cnt = 0;
        for(int i = 0; i < n; i++){
            if(s[i] == 1){
                cnt++;
            }
        }
        return cnt;
    }
};





    
    

File: meetings/2019-10-13/yunsoo.py
"""
Author    Yunsoo Jung
Date      2019-10-13
URL       https://leetcode.com/problems/magical-string
Approach  String, BruteForce
"""

class Solution:
    def magicalString(self, n):
        s = [1, 2, 2]
        f = [1, 2, 2]
        toggle = 2
        f_idx = 2
        while(len(s) < n):
            toggle = 1 if toggle == 2 else 2
            k = f[f_idx]
            for i in range(k):
                s.append(toggle)
                f.append(toggle)
            f_idx += 1
        cnt = 0
        for i in range(n):
            if s[i] == 1:
                cnt += 1
        return cnt
